src/signers.ts

```diff
@@ -1,19 +1,19 @@
+import { Signer } from "ethers"
 import type { HardhatRuntimeEnvironment } from "hardhat/types"
-import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
 
 type NamedSigners = {
-  [name: string]: SignerWithAddress
+  [name: string]: Signer
 }
 
 export interface HardhatSignersHelpers {
   getNamedSigners(): Promise<NamedSigners>
-  getUnnamedSigners(): Promise<SignerWithAddress[]>
+  getUnnamedSigners(): Promise<Signer[]>
```

Comment:
In ethers v6 `SignerWithAddress` was migrated to [`HardhatEthersSigner`](https://github.com/NomicFoundation/hardhat/blob/b626dc15b047773603964a66f7efa9982d188bda/packages/hardhat-ethers/src/signers.ts#L20).
Could we use this type instead of the `Signer` interface?

Reply:
Oh you're right, it's even [aliased](https://github.com/NomicFoundation/hardhat/blob/b626dc15b047773603964a66f7efa9982d188bda/packages/hardhat-ethers/src/signers.ts#L266) as `SignerWithAddress`

Reply:
180ac30